adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,34 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source){
+	std::stack<char> stack;
+	auto find_pair = [](char c){
+		for (auto it = brackets.cbegin() ; it != brackets.cend(); it++) {
+		if (c == it->first || c == it->second)
+			return it;
+		}
+		return brackets.cend();
+	};
+	
+	for (auto c : source){
+		auto check = find_pair(c);
+		if (check != brackets.cend())
+		{
+			if (check->first == c)
```

Comment:
What is check and what is first? Any reader will not understand this, so please give your conditions a name with const bool flags.